axelor-account/src/main/resources/views/InvoiceTerm.xml

```diff
@@ -154,4 +281,36 @@
     <call class="com.axelor.apps.account.web.InvoiceTermController"
       method="initInvoiceTermFromInvoice"/>
   </action-method>
+
+  <action-method name="action-invoice-term-method-refusal-to-pay">
+    <call class="com.axelor.apps.account.web.InvoiceTermController" method="refusalToPay"/>
+  </action-method>
+
+  <action-method name="action-invoice-term-method-domain-pfp-validator-user-on-select">
+    <call class="com.axelor.apps.account.web.InvoiceTermController"
+      method="setPfpValidatorUserDomain"/>
+  </action-method>
+
+  <action-method name="action-invoice-term-method-hide-send-email-pfp-btn">
+    <call class="com.axelor.apps.account.web.InvoiceTermController"
+      method="hideSendEmailPfpBtn"/>
+  </action-method>
+
+  <action-view name="action-invoiceterm-view-refusal-to-pay" title="Cancellation"
+    model="com.axelor.apps.account.db.InvoiceTerm">
+    <view type="form" name="invoice-term-refusal-to-pay-wizard-form"/>
+    <view-param name="popup" value="reload"/>
+    <view-param name="show-toolbar" value="false"/>
+    <view-param name="show-confirm" value="false"/>
+    <view-param name="popup-save" value="false"/>
+    <view-param name="forceEdit" value="true"/>
+    <context name="_showRecord" expr="eval: __this__.id"/>
+    <context name="_xApplicationType" expr="eval: _model"/>
+  </action-view>
+
+  <action-condition name="action-invoiceterm-refusal-to-pay-cancel-reason-check">
+    <check error="A cancel reason must be selected" field="cancelReason"
+      if="reasonOfRefusalToPay == null || reasonOfRefusalToPay == 0"/>
+  </action-condition>
```

Comment:
Please make the field required instead and add a check in java method on the field to avoid a NPE in case of UI error.